diff --git a/t/t5300-pack-object.sh b/t/t5300-pack-object.sh
index 73445782e7..ffdc9fa954 100755
--- a/t/t5300-pack-object.sh
+++ b/t/t5300-pack-object.sh
@@ -118,7 +118,7 @@ test_expect_success 'pack-object <stdin parsing: [|--revs] with --stdin' '
 	test_cmp err.expect err.actual
 '
 
-test_expect_success 'pack-object <stdin parsing: --stdin-packs handles garbage' '
+test_expect_failure 'pack-object <stdin parsing: --stdin-packs handles garbage' '
 	cat >in <<-EOF &&
 	$(git -C pack-object-stdin rev-parse one)
 	$(git -C pack-object-stdin rev-parse two)
